oqsprov/oqsprov.c

```diff
@@ -900,6 +900,23 @@ static const OSSL_DISPATCH oqsprovider_dispatch_table[]
 #    define OQS_PROVIDER_ENTRYPOINT_NAME OSSL_provider_init
 #endif // ifdef OQS_PROVIDER_STATIC
 
+static CRYPTO_ONCE once_obj_create = CRYPTO_ONCE_STATIC_INIT;
+static CRYPTO_RWLOCK *lock_obj_create;
+
+/* Initializes a lock for `OBJ_create`. */
+static void init_obj_create_lock(void)
+{
+    lock_obj_create = CRYPTO_THREAD_lock_new();
+}
+
+#ifndef _WIN32
```

Comment:
It looks "fishy" to me that this exit routine is guarded against "WIN32": How/when is this unlock then done under Windows? Why don't I find this routine elsewhere in the PR?